diff --git a/simulation.go b/simulation.go
index 0a8bd01..9aa0f44 100644
--- a/simulation.go
+++ b/simulation.go
@@ -15,8 +15,11 @@ import (
 	"github.com/dedis/kyber/util/key"
 	"github.com/dedis/student_18_dgcosi/onet/log"
 	"github.com/dedis/student_18_dgcosi/onet/network"
+	"github.com/dedis/student_18_dgcosi/dg-cosi/crypto"
 )
 
+var DGENABLE = false
+
 type simulationCreate func(string) (Simulation, error)
 
 var simulationRegistered map[string]simulationCreate
@@ -242,9 +245,16 @@ func (s *SimulationBFTree) CreateRoster(sc *SimulationConfig, addresses []string
 	entities := make([]*network.ServerIdentity, hosts)
 	log.Lvl3("Doing", hosts, "hosts")
 	key := key.NewKeyPair(suite)
+	dgKey := crypto.ConvertNormalKeyToDg(key.Private, suite)
 	for c := 0; c < hosts; c++ {
-		key.Private.Add(key.Private, suite.Scalar().One())
-		key.Public.Add(key.Public, suite.Point().Base())
+		if DGENABLE{
+			dgKey.NextKey(suite)
+			key.Public = dgKey.Public
+			key.Private = dgKey.Private.X
+		} else {
+			key.Private.Add(key.Private, suite.Scalar().One())
+			key.Public.Add(key.Public, suite.Point().Base())
+		}
 		address := addresses[c%nbrAddr] + ":"
 		var add network.Address
 		if localhosts {
